### DISCUSSION TITLE
Change in behavior WRT initonly fields

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/runtime#66911] [General]
In .NET 4.8, setting a `readonly` (initonly) field in a method other than the ctor caused `VerificationException` (say, when you use IL emit). In .NET 6, it succeeds and the value is set. Is that an expected change?

**Comment by reflectronic** [comment]
.NET Core does not (and has never) performed IL verification under any circumstances. It is expected that unverifiable IL will compile and run without issue.

**Reply by TahirAhmadov** [reply]
OK so it's a known change from .NET FW? Thanks.

**Reply by danmoseley** [reply]
@steveharter afaik we haven't removed any error checking from ref emit API?
